packages/babel-plugin-jest-hoist/src/index.ts

```diff
@@ -152,38 +164,113 @@ FUNCTIONS.deepUnmock = args => args.length === 1 && args[0].isStringLiteral();
 FUNCTIONS.disableAutomock = FUNCTIONS.enableAutomock = args =>
   args.length === 0;
 
-export default (): {visitor: Visitor} => {
-  const shouldHoistExpression = (expr: NodePath): boolean => {
-    if (!expr.isCallExpression()) {
-      return false;
-    }
+const createJestObjectGetter = statement`
+function GETTER_NAME() {
+  const { JEST_GLOBALS_MODULE_JEST_EXPORT_NAME } = require("JEST_GLOBALS_MODULE_NAME");
+  GETTER_NAME = () => JEST_GLOBALS_MODULE_JEST_EXPORT_NAME;
+  return JEST_GLOBALS_MODULE_JEST_EXPORT_NAME;
+}
+`;
 
-    // TODO: avoid type casts - the types can be arrays (is it possible to ignore that without casting?)
-    const callee = expr.get('callee') as NodePath;
-    const expressionArguments = expr.get('arguments');
-    const object = callee.get('object') as NodePath;
-    const property = callee.get('property') as NodePath;
-    return (
-      property.isIdentifier() &&
-      FUNCTIONS[property.node.name] &&
-      (object.isIdentifier(JEST_GLOBAL) ||
-        (callee.isMemberExpression() && shouldHoistExpression(object))) &&
-      FUNCTIONS[property.node.name](
-        Array.isArray(expressionArguments)
-          ? expressionArguments
-          : [expressionArguments],
-      )
-    );
-  };
-
-  const visitor: Visitor = {
-    ExpressionStatement(path) {
-      if (shouldHoistExpression(path.get('expression') as NodePath)) {
-        // @ts-ignore: private, magical property
-        path.node._blockHoist = Infinity;
-      }
-    },
-  };
+const isIdentifierJestObject = (identifier: NodePath<Identifier>): boolean => {
+  // global
+  if (
+    identifier.node.name === JEST_GLOBAL_NAME &&
+    !identifier.scope.hasBinding(JEST_GLOBAL_NAME)
+  ) {
+    return true;
+  }
+  // import from '@jest/globals'
+  if (
+    identifier.referencesImport(
```

Comment:
woah, so nice!

Reply:
I'm afraid it might not work for import namespace specifiers though (although IMO it should, in simple cases). But can add another condition for that.

Reply:
Unless I misunderstand it does?

https://github.com/babel/babel/blob/7e6839f42b9872e2f3d4925e5e41876fcec9b3ad/packages/babel-traverse/src/path/introspection.js#L185-L187

Reply:
For import namespace seems it just works if you ask it if the identifier is the whole thing ('*') - not very useful :sweat_smile: https://github.com/babel/babel/blob/7e6839f42b9872e2f3d4925e5e41876fcec9b3ad/packages/babel-traverse/src/path/introspection.js#L181-L183